src/main/java/com/sysc4806app/services/UserService.java

```diff
@@ -73,4 +73,35 @@ private long getEuclideanDistanceLHSPrebuilt(HashMap<Product, Review> lhs_map, U
         }
         return score;
     }
+
+    public int getDegreeOfSeparationNumber(User root, User target,Collection<User> visited){
```

Comment:
Needing to pass it an empty ArrayList is a bit weird. You could add a wrapper function that takes care of that for you. Up to you.

Reply:
good tip, ill add it right away